File: CucumberJVM/Chapter13/13.1-LocalScenarioContextPattern/After/src/main/java/com/wimp/app/services/AuthenticationService.java
package com.wimp.app.services;

import com.wimp.app.infrastructure.UserSession;

public class AuthenticationService {
    public static void login(String customerName) {
        UserSession.getCurrent().setAuthenticatedCustomerName(customerName);
    }

    public static boolean isAuthenticated(String customerName) {
        return customerName.equals(UserSession.getCurrent().getAuthenticatedCustomerName());
    }
}

File: Reqnroll/Chapter13/13.1-LocalScenarioContextPattern/Before/WIMP.App/Services/AuthenticationService.cs
using WIMP.App.Infrastructure;

namespace WIMP.App.Services;

public static class AuthenticationService
{
    public static void Login(string customerName)
    {
        UserSession.Current.AuthenticatedCustomerName = customerName;
    }

    public static bool IsAuthenticated(string? customerName)
    {
        return customerName != null && UserSession.Current.AuthenticatedCustomerName == customerName;
    }
}
